lib/source-code/source-code.js

```diff
@@ -411,19 +411,29 @@ class SourceCode extends TokenStore {
     }
 
     /**
-     * Determines if two tokens have at least one whitespace character
-     * between them. This completely disregards comments in making the
-     * determination, so comments count as zero-length substrings.
-     * @param {Token} first The token to check after.
-     * @param {Token} second The token to check before.
-     * @returns {boolean} True if there is only space between tokens, false
-     *  if there is anything other than whitespace between tokens.
+     * Determines if two nodes or tokens have at least one whitespace character
+     * between them.
+     * @param {ASTNode|Token} first The node or token to check after.
+     * @param {ASTNode|Token} second The node or token to check before.
+     * @returns {boolean} True if there is a whitespace character between
+     * any of the tokens found between the two given nodes or tokens.
      * @public
      */
     isSpaceBetweenTokens(first, second) {
-        const text = this.text.slice(first.range[1], second.range[0]);
+        const finalToken = (this.getFirstToken(second) || second);
+        let currentToken = this.getLastToken(first) || first;
 
-        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
+        while (currentToken !== finalToken) {
+            const nextToken = this.getTokenAfter(currentToken, { includeComments: true });
+
+            if (/\s/u.test(this.text.slice(currentToken.range[1], nextToken.range[0]))) {
```

Comment:
Could this be replaced with `if (currentToken.range[1] !== nextToken.range[0])`? I.e. is there a situation where something else other than whitespace can be between two consecutive tokens? I don't know, just asking.

Reply:
Yeah, I think that would work. Given that it's including comments, I don't think there can be anything but whitespace.

Reply:
Done!